src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -641,12 +664,12 @@ private void CompileNamedType(NamedTypeSymbol containingType)
             compilationState.Free();
         }
 
-        private void CompileSynthesizedMethods(PrivateImplementationDetails privateImplClass, DiagnosticBag diagnostics)
+        private void CompileSynthesizedMethods(Cci.INamespaceTypeDefinition synthesizedTopLevelType, DiagnosticBag diagnostics)
         {
             Debug.Assert(_moduleBeingBuiltOpt != null);
 
             var compilationState = new TypeCompilationState(null, _compilation, _moduleBeingBuiltOpt);
-            foreach (MethodSymbol method in privateImplClass.GetMethods(new EmitContext(_moduleBeingBuiltOpt, null, diagnostics, metadataOnly: false, includePrivateMembers: true)))
+            foreach (MethodSymbol method in synthesizedTopLevelType.GetMethods(new EmitContext(_moduleBeingBuiltOpt, null, diagnostics, metadataOnly: false, includePrivateMembers: true)))
```

Comment:
>synthesizedTopLevelType.GetMethods [](start = 44, length = 34)

Does this API return a MethodSymbol? Is the explicit cast safe?

Reply:
Please see my comment that calls this out: https://github.com/dotnet/roslyn/pull/43281#discussion_r407011548

Reply:
> Please see my comment that calls this out: [https://github.com/dotnet/roslyn/pull/43281#discussion_r407011548](https://github.com/dotnet/roslyn/pull/43281#discussion_r407011548)

Whatever we are doing for PrivateImplementationDetails is not applicable in general case. So, I think the generalization of this helper is not the right thing to do.

---
In reply to: [409497490](https://github.com/dotnet/roslyn/pull/43281#discussion_r409497490) [](ancestors = 409497490)